react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+
+class ValidateInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      entry: 'start'
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    if (event.target.value.length >= 8) {
+      this.setState({
+        entry: 'pass'
+      });
+    } else if (event.target.value.length === 0) {
+      this.setState({
+        entry: 'empty'
+      });
+    } else if (event.target.value.length < 8) {
+      this.setState({
+        entry: 'short'
+      });
+    }
+  }
+
+  render() {
+    if (this.state.entry === 'empty') {
+      return (
+        <form className="container">
+          <label>Password</label>
+          <div className="input-container">
+            <input type="password" className="input" onChange={this.handleChange} />
```

Comment:
Be sure to always pass a `value` prop to your input as well so that it is fully "controlled".